File: src/main/java/me/zeroeightsix/kami/module/modules/gui/ArmourHide.java
package me.zeroeightsix.kami.module.modules.gui;

import me.zeroeightsix.kami.module.Module;
import me.zeroeightsix.kami.setting.Setting;
import me.zeroeightsix.kami.setting.Settings;
import net.minecraft.inventory.EntityEquipmentSlot;

/**
 * Created by TBM on 30/12/2019.
 */
@Module.Info(name = "ArmourHide", category = Module.Category.GUI, description = "Hides entity armour", showOnArray = Module.ShowOnArray.OFF)
public class ArmourHide extends Module {
    public Setting<Boolean> player = register(Settings.b("Players", false));
    public Setting<Boolean> armourstand = register(Settings.b("Armour Stands", true));
    public Setting<Boolean> mobs = register(Settings.b("Mobs", true));

    public Setting<Boolean> helmet = register(Settings.b("Helmet", false));
    public Setting<Boolean> chestplate = register(Settings.b("Chestplate", false));
    public Setting<Boolean> leggins = register(Settings.b("Leggings", false));
    public Setting<Boolean> boots = register(Settings.b("Boots", false));

    public static ArmourHide INSTANCE;

    public ArmourHide() {
        ArmourHide.INSTANCE = this;
    }

    public static boolean shouldRenderPiece(EntityEquipmentSlot slotIn) {
        if (slotIn == EntityEquipmentSlot.HEAD && ArmourHide.INSTANCE.helmet.getValue()) {
            return true;
        } else if (slotIn == EntityEquipmentSlot.CHEST && ArmourHide.INSTANCE.chestplate.getValue()) {
            return true;
        } else if (slotIn == EntityEquipmentSlot.LEGS && ArmourHide.INSTANCE.leggins.getValue()) {
            return true;
        } else if (slotIn == EntityEquipmentSlot.FEET && ArmourHide.INSTANCE.boots.getValue()) {
            return true;
        } else {
            return false;
        }
    }
}

File: src/main/java/me/zeroeightsix/kami/module/modules/render/ArmourHide.kt
package me.zeroeightsix.kami.module.modules.render

import me.zeroeightsix.kami.module.Module
import me.zeroeightsix.kami.setting.Settings
import net.minecraft.inventory.EntityEquipmentSlot

@Module.Info(
        name = "ArmourHide",
        category = Module.Category.RENDER,
        description = "Hides the armour on selected entities",
        showOnArray = Module.ShowOnArray.OFF
)
object ArmourHide : Module() {
    val player = register(Settings.b("Players", false))
    val armourStand = register(Settings.b("ArmourStands", true))
    val mobs = register(Settings.b("Mobs", true))
    private val helmet = register(Settings.b("Helmet", false))
    private val chestplate = register(Settings.b("Chestplate", false))
    private val leggings = register(Settings.b("Leggings", false))
    private val boots = register(Settings.b("Boots", false))

    @JvmStatic
    fun shouldHidePiece(slotIn: EntityEquipmentSlot): Boolean {
        return helmet.value && slotIn == EntityEquipmentSlot.HEAD
                || chestplate.value && slotIn == EntityEquipmentSlot.CHEST
                || leggings.value && slotIn == EntityEquipmentSlot.LEGS
                || boots.value && slotIn == EntityEquipmentSlot.FEET
    }
}